### PR TITLE
React-app: Handle routes on the client itself instead of `ui.go`

### LINKED ISSUE #3216: Remove the need for specifying React UI's paths in `ui.go`
We are using client-side routing in the new React UI. This means that for all the client-side routes (ex. `/new/graph`, `/new/stores`, etc.) in the new UI, we need to serve the same `index.html`. Right now we maintain an array of paths in the Go code, named `reactAppPaths`. If the incoming request's path is found in this array, we directly serve the `index.html` file, otherwise, we return `404` or a file if one exists with the given path.

Instead of maintaining the array, we can replace `404` with the `index.html` file. This is a common model in serving single-page applications. If the incoming request corresponds to a static asset, return that asset, otherwise return the `index.html` file. If the given route doesn't exist, then we can show a 404 page directly on the client-side, using a default route in React.

### PR DESCRIPTION
**Pull request by daksh-sagar**
Fixes #3216 

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Remove `reactPaths` from ui.go
Create `NotFound` component in the react-app and render it when a route is not found.

### CODE REVIEW

**Comment by daksh-sagar** [comment]
@prmsrswt

File: pkg/ui/react-app/src/pages/notFound/NotFound.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React, { FC } from 'react';
```
Comment (prmsrswt) [inline pkg/ui/react-app/src/pages/notFound/NotFound.tsx:1]: Can you move this component to `src/thanos/pages`? All of our custom pages are there.

Comment (daksh-sagar) [reply pkg/ui/react-app/src/pages/notFound/NotFound.tsx:1]: Ofcourse !
Is everything else fine ?

Comment (prmsrswt) [reply pkg/ui/react-app/src/pages/notFound/NotFound.tsx]: Yep, everything else seems to be alright. I'll test it out locally first and then let's get this merged.

**Comment by prmsrswt** [comment]
Sorry for not responding fast enough. We already have these changes in the codebase so I am going to close this PR. That being said we still appreciate your effort (very much) and will try to not waste it like this in the future.

**Pull request closed without merging**